layer-1-contracts/README.md

```diff
@@ -0,0 +1,19 @@
+# Layer 1 Contracts
+
+## Design Goals
+We want our ORU contracts to be built in compliance with the following goals:
+- A **pluggable state machine** which:
+    - allows separation of concerns and parallelizable work
+    - is fully stateless, i.e. all state transitions can be expressed in the form `bytes32 _inputHash, bytes _witness => bytes32 _outputhash` and are only dependent on the immediately preceding pre-state
```

Comment:
cc @protolambda , this is an explicit callout that steps are only dependent on the immediately preceding one